docs/source/amp.rst

```diff
@@ -1,218 +1,22 @@
 .. role:: hidden
     :class: hidden-section
 
-Automatic Mixed Precision package - torch.cuda.amp
+Automatic Mixed Precision package - torch.autocast
```

Comment:
The name here is confusing, because torch.autocast doesn't actually exist as a module. Did you mean `autocast_mode`?

Reply:
But tbh, torch.autocast is a way better name than torch.autocast_mode... can we change it? :)

Reply:
Thanks for the suggestions. I think `amp` should be the package name as the convention. So I move `autocast_mode.py` into the `amp` package.